File: 0129-sum-root-to-leaf-numbers/solution.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
void sumPath(struct TreeNode* root, int res, int *psum){
    if(root == NULL) return ;
    if(!root->left && !root->right) *psum += res*10+root->val;
    sumPath(root->left, res*10+root->val,psum);
    sumPath(root->right, res*10+root->val,psum);
}
int sumNumbers(struct TreeNode* root) {
    int sum = 0;
    sumPath(root,0,&sum);
    return sum;
}

File: 0129-sum-root-to-leaf-numbers/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private  void dfs(TreeNode root, int res){
        if(root == null) return;
        // System.out.println(res + " "  + sum);
        if(root.left == null && root.right == null){
            sum += (res * 10) + root.val;
        }
        dfs(root.left,res*10+root.val);
        dfs(root.right,res*10+root.val);
    }
    private int sum = 0;
    public int sumNumbers(TreeNode root) {
        dfs(root, 0);
        return sum;
    }
}
